rs3_trading/historicaldata/historical_process.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+import json
+import logging
+import urllib
+from collections import deque
+from time import sleep
+from typing import Any, Coroutine, Iterable
+
+import pandas as pd
+import requests
+
+from rs3_trading.database.dbcm import DuckDBCM
+from rs3_trading.historicaldata.historical_utils import extract_item_name_from_table, historical_data_transformation
+from rs3_trading.historicaldata.request_errors import check_request_status, NotFoundError, RequestTimeoutError, TooManyRequestsError
+from rs3_trading.utils.utilities import create_database, insert_into_table, join_with_commas
+
+DATABASE_NAME = 'GE_Tick_Data.db'
+COLUMN_NAME_ITEM_NAME = 'item_name'
+LOG_FILE = 'historicaldata_log.log'
+
+BATCH_SIZE = 5
+
+API_URL = 'https://api.weirdgloop.org/exchange/history/rs/last90d?name='
+
+
+
+query_string = f'select {COLUMN_NAME_ITEM_NAME} from rs3_price_data'
+logging.basicConfig(filename=LOG_FILE, level=logging.ERROR, format='%(asctime)s - %(levelname)s - %(message)s')
+
+item_names: deque = extract_item_name_from_table(query_string, DATABASE_NAME)
```

Comment:
This is also a bit of shared state, I would find a way to move it into your historical main and pass it as a parameter